docs/csharp/tour-of-csharp/classes-and-objects.md

```diff
@@ -0,0 +1,725 @@
+		}
+	}
+}
+```
+
+C# requires a local variable to be ***definitely assigned*** before its value can be obtained. For example, if the declaration of the previous i did not include an initial value, the compiler would report an error for the subsequent usages of i because i would not be definitely assigned at those points in the program.
+
+A method can use `return` statements to return control to its caller. In a method returning `void`, `return` statements cannot specify an expression. In a method returning non-void, `return` statements must include an expression that computes the return value.
+
+### Static and instance methods
+
+A method declared with a static modifier is a ***static method***. A static method does not operate on a specific instance and can only directly access static members.
+
+A method declared without a static modifier is an ***instance method***. An instance method operates on a specific instance and can access both static and instance members. The instance on which an instance method was invoked can be explicitly accessed as this. It is an error to refer to this in a static method.
+
+The following Entity class has both static and instance members.
+
+```csharp
+class Entity
+{
+	static int nextSerialNo;
+	int serialNo;
+	public Entity() 
+	{
+		serialNo = nextSerialNo++;
+	}
+	public int GetSerialNo() 
+	{
+		return serialNo;
+	}
+	public static int GetNextSerialNo() 
+	{
+		return nextSerialNo;
+	}
+	public static void SetNextSerialNo(int value) 
+	{
+		nextSerialNo = value;
+	}
+}
+```
+
+Each `Entity` instance contains a serial number (and presumably some other information that is not shown here). The `Entity` constructor (which is like an instance method) initializes the new instance with the next available serial number. Because the constructor is an instance member, it is permitted to access both the `serialNo` instance field and the `nextSerialNo` static field.
+
+The `GetNextSerialNo` and `SetNextSerialNo` static methods can access the `nextSerialNo` static field, but it would be an error for them to directly access the `serialNo` instance field.
+
+The following example shows the use of the Entity class.
+
+```csharp
+using System;
+class EntityExample
+{
+	static void Main() 
+	{
+		Entity.SetNextSerialNo(1000);
+		Entity e1 = new Entity();
+		Entity e2 = new Entity();
+		Console.WriteLine(e1.GetSerialNo());            // Outputs "1000"
+		Console.WriteLine(e2.GetSerialNo());            // Outputs "1001"
+		Console.WriteLine(Entity.GetNextSerialNo());    // Outputs "1002"
+	}
+}
+```
+
+Note that the `SetNextSerialNo` and `GetNextSerialNo` static methods are invoked on the class whereas the `GetSerialNo` instance method is invoked on instances of the class.
+
+### Virtual, override, and abstract methods
+
+When an instance method declaration includes a `virtual` modifier, the method is said to be a ***virtual method***. When no virtual modifier is present, the method is said to be a ***non-virtual method***.
```

Comment:
you use nonvirtual and non-virtual in this topic. I've consulted Microsoft's style guide and we should only hyphenate words beginning with non if necessary to avoid confusion.